test/byContract/IexecAccessors/IexecAccessors.test.ts

```diff
@@ -78,11 +84,89 @@ describe('IexecAccessors', async () => {
         expect(await iexecPoco.totalSupply()).to.equal(0n);
     });
 
+    it('balanceOf', async function () {
+        const amount = 3;
+        await iexecWrapper.depositInIexecAccount(anyone, amount);
+        expect(await iexecPoco.balanceOf(anyone.address)).to.equal(amount);
+    });
+
+    it('frozenOf', async function () {
+        await createDeal(); // Lock some requester funds.
+        const dealPrice = appPrice + datasetPrice + workerpoolPrice; // volume == 1
+        expect(await iexecPoco.frozenOf(requester.address)).to.equal(dealPrice);
+    });
+
+    it('allowance', async function () {
+        const amount = 10;
+        const spender = ethers.Wallet.createRandom().address;
+        await iexecWrapper.depositInIexecAccount(anyone, amount);
+        await iexecPoco.connect(anyone).approve(spender, amount);
+        expect(await iexecPoco.allowance(anyone.address, spender)).to.equal(amount);
+    });
+
+    it('viewAccount', async function () {
+        await createDeal(); // Lock some requester funds.
+        const dealPrice = appPrice + datasetPrice + workerpoolPrice;
+        // Stake some funds.
+        const stakedBalance = 3;
+        await iexecWrapper.depositInIexecAccount(requester, stakedBalance);
+        // Check staked and locked amounts.
+        const account = await iexecPoco.viewAccount(requester.address);
+        expect(account.stake).to.equal(stakedBalance);
+        expect(account.locked).to.equal(dealPrice);
+    });
+
     // TODO test the case where token() == 0x0 in native mode.
     it('token', async function () {
         expect(await iexecPoco.token()).to.equal('0x5FbDB2315678afecb367f032d93F642f64180aa3');
     });
 
+    it('viewDeal', async function () {
+        const { dealId } = await createDeal();
+        const deal = await iexecPoco.viewDeal(dealId);
+        console.log('🚀 ~ deal sponsor:', deal.sponsor);
+        expect(deal.app.pointer).to.equal(appAddress);
+        expect(deal.app.owner).to.equal(appProvider.address);
+        expect(deal.app.price).to.equal(appPrice);
+        expect(deal.dataset.pointer).to.equal(datasetAddress);
+        expect(deal.dataset.owner).to.equal(datasetProvider.address);
+        expect(deal.dataset.price).to.equal(datasetPrice);
+        expect(deal.workerpool.pointer).to.equal(workerpoolAddress);
+        expect(deal.workerpool.owner).to.equal(scheduler.address);
+        expect(deal.workerpool.price).to.equal(workerpoolPrice);
+        expect(deal.trust).to.equal(1);
+        expect(deal.category).to.equal(0);
+        expect(deal.tag).to.equal(HashZero); // Standard
+        expect(deal.requester).to.equal(requester.address);
+        expect(deal.beneficiary).to.equal(AddressZero);
```

Comment:
Why not:
```suggestion
        expect(deal.trust).to.equal(1);
        expect(deal.category).to.equal(2);
        expect(deal.tag).to.equal(0x..3); // Standard
        expect(deal.requester).to.equal(requester.address);
        expect(deal.beneficiary).to.equal(0x..4);
        [...]
```
?

Reply:
My goal is to use minimal setup code and check that all fields of the deal are returned and not necessarily go into more details because we already have other tests that check all relevant fields for each type of deal.